include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,89 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class RandomAccessIterator>
+auto sorted_mode(RandomAccessIterator first, RandomAccessIterator last) -> std::vector<typename std::iterator_traits<RandomAccessIterator>::value_type>
+{
+    using Z = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
+    using Size = typename std::iterator_traits<RandomAccessIterator>::difference_type;
+
+    std::vector<Z> sorted(first, last);
+    std::vector<Z> modes {};
+    Size max_counter {0};
+    Z previous_search {std::numeric_limits<Z>::max()};
+
+    if(sorted.front() == sorted.back())
+    {
+        modes.push_back(sorted.front());
+        return modes;
+    }
+    
+    for(auto it {sorted.begin()}; it != sorted.end(); ++it)
+    {
+        if(*it == previous_search)
+        {
+            continue;
+        }
+        
+        Size current_count {0};
+        auto end_it {it};
+        while(*end_it == *it)
+        {
+            ++current_count;
+
+            if(end_it == sorted.end())
+            {
+                break;
+            }
+
+            ++end_it;
+        }
+
+        if(current_count > max_counter)
+        {
+            modes.resize(1);
+            modes[0] = *it;
+            max_counter = current_count;
+        }
+
+        else if(current_count == max_counter)
+        {
+            modes.push_back(*it);
+        }
+
+        previous_search = *it;
+    }
+
+    if(max_counter == 1) // Return an empty vector if mode is undefined
+    {
+        modes.clear();
+    }
+
+    return modes;
+}
+
+template<class RandomAccessContainer>
+inline auto sorted_mode(RandomAccessContainer & v)
```

Comment:
This can be marked `const`: `RandomAccessContainer const & v`.

BTW does this require `RandomAccess` or can it be a `std::list`?

Reply:
Now that it is not sorting it can take a forward iterator instead of random access iterator